### ISSUE TITLE
UnSAM notebook not working

### ISSUE THREAD

**Issue by boumpteryx** [frank-xwang/UnSAM#1]
The UnSAM notebook does not run, it is apparently missing some imports (argparse and DefaultPredictor at least)

**Comment by frank-xwang** [comment]
Hi, did you run 

```
python -m pip install 'git+https://github.com/MaureenZOU/detectron2-xyz.git'
python -m pip install -r requirements.txt
```

The detectron2's DefaultPredictor and argparse should be installed if you have completed these installation steps. 

Let me know if this does not fix your issues. :)

**Comment by frank-xwang** [comment]
ps: if you were running notebook locally, please follow `INSTALL.md` to prepare the environment first. If you are using colab, simply compiling all cells should work.

**Comment by frank-xwang** [comment]
Closing it for now. Please feel free to re-open it if you have more questions.

**Comment by abrichr** [comment]
Currently, cell 4 in the notebook raises an exception:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-4-c3af9a589b50>](https://localhost:8080/#) in <cell line: 2>()
      1 import argparse
----> 2 from detectron2.engine import DefaultPredictor, default_setup
      3 from detectron2.projects.deeplab import add_deeplab_config
      4 from detectron2.config import get_cfg
      5 from detectron2.utils.colormap import random_color

7 frames
[/usr/local/lib/python3.10/dist-packages/detectron2/data/transforms/transform.py](https://localhost:8080/#) in ExtentTransform()
     44     """
     45 
---> 46     def __init__(self, src_rect, output_size, interp=Image.LINEAR, fill=0):
     47         """
     48         Args:

AttributeError: module 'PIL.Image' has no attribute 'LINEAR'
```

One possible solution from from https://github.com/facebookresearch/detectron2/issues/5010 is:

```
pip install pillow==9.5.0
```
However:
```
Requirement already satisfied: pillow==9.5.0 in /usr/local/lib/python3.10/dist-packages (9.5.0)
```

Another solution is:

```
python3 -m pip install -U 'git+https://github.com/facebookresearch/detectron2.git@ff53992b1985b63bd3262b5a36167098e3dada02'
```

But in this notebook we are using https://github.com/MaureenZOU/detectron2-xyz.git, so this does not seem applicable.

As a workaround I added this to the top of the relevant cell:

```
import PIL.Image as Image
Image.LINEAR = Image.BILINEAR
```

This resolves that issue, but raises another one when importing `mask2former`:

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[/content/UnSAM/whole_image_segmentation/mask2former/modeling/pixel_decoder/ops/functions/ms_deform_attn_func.py](https://localhost:8080/#) in <module>
     21 try:
---> 22     import MultiScaleDeformableAttention as MSDA
     23 except ModuleNotFoundError as e:

ModuleNotFoundError: No module named 'MultiScaleDeformableAttention'

During handling of the above exception, another exception occurred:

ModuleNotFoundError                       Traceback (most recent call last)
7 frames
[/content/UnSAM/whole_image_segmentation/mask2former/modeling/pixel_decoder/ops/functions/ms_deform_attn_func.py](https://localhost:8080/#) in <module>
     28     )
     29     if torch.cuda.is_available():
---> 30       raise ModuleNotFoundError(info_string)
     31     else:
     32       print("if you are running on GPU", info_string)

ModuleNotFoundError: 

Please compile MultiScaleDeformableAttention CUDA op with the following commands:
	`cd mask2former/modeling/pixel_decoder/ops`
	`sh make.sh`
```

I added this cell immediately above:
```
!git clone https://github.com/facebookresearch/Mask2Former.git
%cd Mask2Former
!pip install -r requirements.txt
%cd mask2former/modeling/pixel_decoder/ops
!sh make.sh
%cd ../../../../..
import sys
sys.path.append('/content/UnSAM/whole_image_segmentation/Mask2Former/')
```

After restarting the runtime, the subsequent cell ran successfully.

So to summarize, the following needs to be added to a cell immediately before the fourth one:

```
!git clone https://github.com/facebookresearch/Mask2Former.git
%cd Mask2Former
!pip install -r requirements.txt
%cd mask2former/modeling/pixel_decoder/ops
!sh make.sh
%cd ../../../../..

import sys
sys.path.append('/content/UnSAM/whole_image_segmentation/Mask2Former/')

import PIL.Image as Image
Image.LINEAR = Image.BILINEAR
```
